browser/plugins/test/unhandled-rejection.test.js

```diff
@@ -36,13 +37,49 @@ describe('plugin: unhandled rejection', () => {
             expect(report.severity).toBe('error')
             expect(report.unhandled).toBe(true)
             expect(report.exceptions[0].errorClass).toBe('UnhandledRejection')
-            expect(report.exceptions[0].message).toBe('Unhandled promise rejection')
+            expect(report.exceptions[0].message).toBe('Rejection reason was not an Error. See "Promise" tab for more detail.')
             expect(report.severityReason).toEqual({ type: 'unhandledPromiseRejection' })
+            expect(report.metaData.promise['rejection reason']).toEqual('undefined (or null)')
+            plugin.destroy()
             done()
           }
         })
 
-        Promise.reject(null) // eslint-disable-line
+        setTimeout(() => Promise.reject(null), 0) // eslint-disable-line
+      })
+
+      it('works with DOMExceptions', done => {
+        const client = new Client(VALID_NOTIFIER)
+        client.configure({ apiKey: 'API_KEY_YEAH' })
+        client.use(plugin)
+        client.transport({
+          sendReport: (logger, config, payload) => {
+            const report = payload.events[0].toJSON()
+            expect(report.severity).toBe('error')
+            expect(report.unhandled).toBe(true)
+            expect(report.exceptions[0].errorClass).toBe('UnhandledRejection')
+            expect(report.exceptions[0].message).toBe('Rejection reason was not an Error. See "Promise" tab for more detail.')
+            expect(report.severityReason).toEqual({ type: 'unhandledPromiseRejection' })
+            expect(report.metaData.promise['rejection reason']).toEqual({
+              DOMException: {
+                name: 'AbortError',
```

Comment:
Should errorClass not be AbortError in this case? Message could be message from the DOMException too. I guess you did this to highlight why it has no stacktrace? I've seen a few dashboards that have a lot of UnhandledRejections in them and so anything we can do to distinguish them in the list view would be good.

Reply:
Good point. The only downside I see of this is it would loose the call to action to look at the "promise" tab, which may or may not contain a "code" property too. But having the more descriptive name/message is probs the better tradeoff.

Reply:
We could put something in the dashboard panel that talks about no stacktraces on javascript projects to explain that and get the best of both worlds